source/server-selection/tests/README.rst

```diff
@@ -65,3 +65,43 @@ the TopologyDescription. Each YAML file contains a key for these stages of serve
 Drivers implementing server selection MUST test that their implementation
 correctly returns the set of servers in ``in_latency_window``. Drivers SHOULD also test
 against ``suitable_servers`` if possible.
+
+Selection Within Latency Window Tests
+>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+
+Each YAML file for these tests has the following format:
+
+- ``topology_description``: the state of a mocked cluster
+
+- ``in_window``: array of servers in the latency window that the selected server
+  is to be chosen from. Each element will have all of the following fields:
+
+  - ``address``: a unique address identifying this server
+
+  - ``active_connection_count``: the number of active connections this server
+    currently has open
+
+  - ``available_connection_count``: the number of available connections this
+    server currently has in its pool.
+
+  - ``wait_queue_length``: the number of requests in the wait queue
+    for this server's pool.
+
+- ``max_pool_size``: the maximum number of connections allowed in a server's
+  connection pool.
+
+- ``expected_frequencies``: a document whose keys are the server addresses from the
+  ``in_window`` array and values are numbers in [0, 1] indicating the frequency
+  at which the server should have been selected.
+
+For each file, pass the information from `in_window` to whatever function is
+used to select a server from within the latency window 1000 times, counting how
+many times each server is selected.  Once 1000 selectoins have been made, verify
+that each server was selected at a frequency within 0.05 of the frequency
```

Comment:
5% tolerance works okay for me in python so far but I'm a little worried about false positives here. For example I've seen the delta reach 4%. So I suspect we might need to raise the tolerance to 6% or 7% (or increase the # of iterations) in the future. Can we increase the # of iterations to 2000 to avoid the need to adjust it later?

Reply:
fixed